### ISSUE TITLE
Deprecate loss_type `margin` with Rasa 3.0

### ISSUE THREAD

**Issue by dakshvar22** [RasaHQ/rasa#7891]
**Description of Problem**: Currently, all ML components can be configured to use either `margin` loss or `cross_entropy`(formerly known as `softmax`) loss. Results have shown that `cross_entropy` loss is much better across various configs and datasets. We should deprecate `margin` loss with Rasa 3.0 .

**Overview of the Solution**: Deprecate `margin` loss and also along with it keep only one `similarity_type` - `inner`. We should check if users are still using `margin` loss and encourage them to switch to `cross_entropy` loss. To do this we should add a warning to the codebase that `margin` loss will be deprecated in the future.

**Comment by TyDunn** [comment]
@dakshvar22 Is this something we need to do for 3.0?

**Comment by dakshvar22** [comment]
There have been some developments after the issue was created to suggest that this shouldn't be done. For example, one customer found `margin` loss useful over `cross_entropy`.

@JEM-Mosig was there a decision made on whether triplet loss should replace margin loss in 3.0 itself?
